### PR TITLE
Development

### PR DESCRIPTION
**Pull request by MKumaran98**

### CODE REVIEW

**Comment by DarkShadow20** [comment]
LGTM

File: Config/Mongoose.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const mongoose=require('mongoose');
+
+// mongoose.connect(process.env["DB_URI"], {useNewUrlParser: true, useUnifiedTopology: true});
```
Comment (poojasabnani) [inline Config/Mongoose.js]: commented out code can be removed

File: Config/Mongoose.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+mongoose.connect('mongodb://localhost/quizez', {useNewUrlParser: true, useUnifiedTopology: true})
+
+const db=mongoose.connection;
+
+db.on('error', console.error.bind("Something went wrong with the Mongodb Server"));
+
+db.once("open",()=>console.log("Connection extablished with database"))
+
+module.exports=db;
```
Comment (poojasabnani) [inline Config/Mongoose.js:11]: exports can be added in an object

File: Config/Mongoose.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const mongoose=require('mongoose');
+
+// mongoose.connect(process.env["DB_URI"], {useNewUrlParser: true, useUnifiedTopology: true});
+mongoose.connect('mongodb://localhost/quizez', {useNewUrlParser: true, useUnifiedTopology: true})
+
+const db=mongoose.connection;
+
+db.on('error', console.error.bind("Something went wrong with the Mongodb Server"));
+
+db.once("open",()=>console.log("Connection extablished with database"))
```
Comment (poojasabnani) [inline Config/Mongoose.js:9]: spelling error can be corrected

File: Config/Passport.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const passport=require('passport');
+const JwtStrategy=require('passport-jwt').Strategy;
+const ExtractJwt=require('passport-jwt').ExtractJwt;
+const usersdb=require('../Models/users.model')
+
+const opts={
+    jwtFromRequest : ExtractJwt.fromAuthHeaderAsBearerToken(),
```
Comment (poojasabnani) [inline Config/Passport.js:7]: this is nice!

File: Controller/Questions.controller.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
+const questionsdb=require('../Models/question.model')
+const quizdb=require('../Models/quiz.model')
+const admindb=require('../Models/admin.model')
+
+module.exports.createQuestion=async (req,res)=>{
+    const {question,options,multipleCorrect,points,negativePoints,quiz:quizId,hint}=req.body
```
Comment (poojasabnani) [inline Controller/Questions.controller.js:6]: can be better formatted for better readability

File: Controller/Questions.controller.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    }catch(error){
+        console.log(error)
+        return res.status(503).json({
+            message:"Error creating Question"
+        })
+    }
+}
+
+module.exports.editQuestion=async (req,res)=>{
+    const {question:content,options,multipleCorrect,points,negativePoints,quiz:quizId,hint,createdBy}=req.body;
```
Comment (poojasabnani) [inline Controller/Questions.controller.js:49]: here as well

File: Controller/Quiz.controller.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
+const quizdb=require('../Models/quiz.model')
+const admindb=require('../Models/admin.model')
```
Comment (poojasabnani) [inline Controller/Quiz.controller.js:2]: index.js can be created to import models

File: Controller/User.controller.js [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+        })
+    }else{
+        return res.status(404).json({
+            ok:false,
+            message:"invalid Email ID"
+        })
+    }
+}
+
+module.exports.signinUser=async (req,res)=>{
```
Comment (poojasabnani) [inline Controller/User.controller.js:85]: can be renamed to LoginUser

File: Controller/User.controller.js [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    try{
+        const user=await usersdb.findOne({email:email})
+        if (!user || !(bcrypt.compareSync(password,user.password))){
+            return res.status(401).json({
+                ok:false,
+                message: "Invalid username or password",
+            });
+        }
+        return res.status(200).json({
+            message: 'Sign in successful, here is your token, please keep it safe!',
```
Comment (poojasabnani) [inline Controller/User.controller.js:96]: minor suggestion, to either use double or single quotes consistently across the code.

File: Middleware/quizCheck.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+const quizdb=require('../Models/quiz.model');
+
+const quizCheck=async (req,res,next)=>{
```
Comment (poojasabnani) [inline Middleware/quizCheck.js:3]: can be renamed to something more meaningful

File: Models/question.model.js [hunk 11]
```diff
@@ -0,0 +1,31 @@
+const {Schema,model}=require('mongoose')
+
+const questionSchema=new Schema({
+    question:{
```
Comment (poojasabnani) [inline Models/question.model.js:4]: formatting can be made better for readability

File: Models/quiz.model.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
+const {Schema,model}=require('mongoose');
+
+const quizSchema=new Schema({
+    name:{
```
Comment (poojasabnani) [inline Models/quiz.model.js:4]: here as well

**Review by poojasabnani — CHANGES_REQUESTED** [review]
Hey Muthu, nicely written code esp with using status codes and bcrypt!
Have suggested some changes wrt consistent code and formatting.
Also, couldn't see the hints currently being used in the app. If not used can be cleaned up if not required.

**Review by DarkShadow20 — COMMENTED** [review]
LGTM

File: Config/Mongoose.js (hunk 1 above)
Comment (MKumaran98) [reply Config/Mongoose.js]: I've updated this

File: Config/Mongoose.js (hunk 2 above)
Comment (MKumaran98) [reply Config/Mongoose.js:11]: There is only one export here

File: Controller/User.controller.js (hunk 8 above)
Comment (MKumaran98) [reply Controller/User.controller.js:85]: I've used Sign-up and Sign-in both here and in FE

File: Controller/Quiz.controller.js (hunk 7 above)
Comment (MKumaran98) [reply Controller/Quiz.controller.js:2]: Sure I'll refactor

File: Controller/User.controller.js (hunk 9 above)
Comment (MKumaran98) [reply Controller/User.controller.js:96]: updated

**Pull request merged**